src/http.rs

```diff
@@ -139,8 +139,12 @@ pub fn build_proxy_connector(
     tls_settings: MaybeTlsSettings,
     proxy_config: &ProxyConfig,
 ) -> Result<ProxyConnector<HttpsConnector<HttpConnector>>, HttpError> {
+    let tls = tls_connector_builder(&tls_settings.clone())
+        .context(BuildTlsConnectorSnafu)?
+        .build();
     let https = build_tls_connector(tls_settings)?;
     let mut proxy = ProxyConnector::new(https).unwrap();
+    proxy.set_tls(Some(tls));
```

Comment:
Could you add a comment above `let https = ...` that describes that the line configures our ability to connect to an HTTP/HTTPS proxy, and a corresponding comment above `proxy.set_tls(...)` that points out that this ensures that TLS settings are applied to the actual _proxied_ connection?